src/common/feedGenerator.ts

```diff
@@ -654,14 +655,43 @@ export const configuredFeedBuilder = (
   return newBuilder;
 };
 
-export const bookmarksFeedBuilder = (
-  ctx: Context,
-  unreadOnly: boolean,
-  listId: string | null,
-  builder: SelectQueryBuilder<Post>,
-  alias: string,
-  query?: string | null,
-): SelectQueryBuilder<Post> => {
+interface BookmarksFeedBuilderProps {
+  ctx: Context;
+  unreadOnly: boolean;
+  listId?: string | null;
+  builder: SelectQueryBuilder<Post>;
+  alias: string;
+  query?: string | null;
+}
+
+interface GetFirstFolderSubQuery {
+  builder: SelectQueryBuilder<Post>;
+  userId: string;
+  alias?: string;
+}
+
+const getFirstFolderSubQuery = ({
```

Comment:
This will execute on a loop right?
Maybe also good opportunity for dataLoader? not urgent imo, but nice addition.

Reply:
We can't since this is a query, not a function to fetch data.

Reply:
Also let's move this to `common` since it not directly related to `feedGenerator`

Reply:
Yes, tried to fetch the ID to make it simpler, but it was a sync function, we'll have to always fetch the first folder from the outside if we want, though there can be cases where we will not use it.